src/coreclr/vm/codeman.cpp

```diff
@@ -1525,7 +1525,16 @@ void EEJitManager::SetCpuInfo()
 
     if (((cpuFeatures & ARM64IntrinsicConstants_Sve) != 0) && CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableArm64Sve))
     {
-        CPUCompileFlags.Set(InstructionSet_Sve);
+        int maxVectorLength = (maxVectorTBitWidth >> 3);
+        uint64_t systemVectorTLength = GetSystemVectorLength();
+
+        if (maxVectorLength >= systemVectorTLength)
```

Comment:
It looks like we're missing the handling to artificially restrict `maxVectorLength` to 128 for the time being, which is needed until the handling for `256-bit` or `512-bit` vectors can be plumbed through the JIT on Arm64 (I don't expect that's a lot of work, since its already done for x64, but I expect there to be some Arm64 specific paths that need adjusted)

Reply:
> handling to artificially restrict maxVectorLength to 128 for the time being

yes, I am hoping that we will do it in this PR as I mentioned in https://github.com/dotnet/runtime/pull/101295#discussion_r1588105539 and do the handling of 256-bit later as a separate PR.

Reply:
Marking this as resolved. https://github.com/dotnet/runtime/pull/101295#issuecomment-2093312795 describes the new behaviour that also handles the default case (maxVectorTBitWidth = 0).